app/resources/v1/document/api.py

```diff
@@ -0,0 +1,140 @@
+    "Document", [
+        "id",
+        "title",
+        "subtitle",
+        "agency",
+        "creators",
+        "report_type",
+        "subjects",
+        "date_published",
+        "year",
+        "start_date",
+        "end_date",
+        "description"
+    ]
+)
+
+
+class DocumentAPI(Resource):
+    def get(self, id):
+        doc = Document(id,
+                       'A Title',
+                       'TYPE',
+                       'A description.')
+        return api_response.success({
+            'document': {
+                'id': doc.id,
+                'title': doc.title,
+                'type': doc.type,
+                'description': doc.description,
+            }
+        })
+
+    @login_required
+    def post(self):
+        try:
+            json = request.get_json(force=True)
+            # TODO: utility for stripping values
+        except Exception as e:
+            return api_response.error(str(e))
+
+        errors = validate_json(json, SCHEMA_PATH, 'submission')
+
+        # validate files
+        if errors.get("files") is None:
+            file_errors = []
+            for file_ in json["files"]:
+                if not os.path.exists(
+                        os.path.join(
+                            current_app.config["UPLOAD_DIRECTORY"],
+                            "some_ID",
+                            secure_filename(file_["name"])
+                        )
+                ):
+                    file_errors.append(
+                        "{} : There was an error submitting this file. Please remove and re-upload.".format(
+                            file_["name"]))
+            if file_errors:
+                errors["files"] = file_errors
+        else:
+            errors["files"] = [
+                "There was an issue submitting your file(s). "
+                "Please make sure you have entered a title for every file."
+            ]
+
+        # validate publication date
+        if errors.get("date_published") is None:
+            # TODO: offset by timezone
+            if datetime.strptime(json.get("date_published"), "%m/%d/%Y") > datetime.now():
+                errors["date_published"] = ["This value exceeds the maximum allowable date."]
+
+        # validate start and end dates
+        if errors.get("start_date") is None and errors.get("end_date") is None:
+            start = json.get("start_date")
+            end = json.get("end_date")
+            if start is not None and end is not None:
+                # TODO: offset by timezone
+                start = datetime.strptime(start, "%m/%d/%Y")
+                end = datetime.strptime(end, "%m/%d/%Y")
+                if start >= end:
+                    errors["start_date"] = ["This date must be earlier than the end date."]
+                    errors["end_date"] = ["This date must be later than the start date."]
+
+        if not errors:
+            # create document
+            doc = Document(1,  # id
```

Comment:
We are currently not committing to the db so a custom formatted `id` is used for testing. This will be left blank when we write to the db.